### ISSUE TITLE
ExpoRandom is undifined

### ISSUE THREAD

**Issue by tdangschulz** [dcangulo/react-native-pkce-challenge#236]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-pkce-challenge@5.1.1` for the project I'm working on.

After upgrading this package to 5.1.1, we have a problem with expo. In your upgrade document, you mention that we should remove the ExpoRandom package. But the check in the generate-random-bytes.native.ts expect if the ExpoModules exists, then assumed the ExpoRandom exists too!

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts b/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
index 31db1d6..1f5f1ac 100644
--- a/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
+++ b/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
@@ -9,15 +9,16 @@ export default function generateRandomBytes(): string {
 
     return bytes;
   }
+ const globelObj = (global as any)
 
-  if ((global as any).ExpoModules) {
-    const bytes = (global as any).ExpoModules.ExpoRandom.getRandomBase64String(BYTE_LENGTH);
+  if (globelObj?.ExpoModules?.ExpoRandom) {
+    const bytes = globelObj.ExpoRandom.getRandomBase64String(BYTE_LENGTH);
 
     return bytes;
   }
 
-  if ((global as any).RNPkceChallenge) {
-    const bytes = (global as any).RNPkceChallenge.randomBytes(BYTE_LENGTH);
+  if (globelObj?.RNPkceChallenge) {
+    const bytes = globelObj.RNPkceChallenge.randomBytes(BYTE_LENGTH);
 
     return bytes;
   }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by elencho** [comment]
> Hi! 👋
> 
> Firstly, thanks for your work on this project! 🙂
> 
> Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-pkce-challenge@5.1.1` for the project I'm working on.
> 
> After upgrading this package to 5.1.1, we have a problem with expo. In your upgrade document, you mention that we should remove the ExpoRandom package. But the check in the generate-random-bytes.native.ts expect if the ExpoModules exists, then assumed the ExpoRandom exists too!
> 
> Here is the diff that solved my problem:
> 
> ```diff
> diff --git a/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts b/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
> index 31db1d6..1f5f1ac 100644
> --- a/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
> +++ b/node_modules/react-native-pkce-challenge/src/generate-random-bytes.native.ts
> @@ -9,15 +9,16 @@ export default function generateRandomBytes(): string {
>  
>      return bytes;
>    }
> + const globelObj = (global as any)
>  
> -  if ((global as any).ExpoModules) {
> -    const bytes = (global as any).ExpoModules.ExpoRandom.getRandomBase64String(BYTE_LENGTH);
> +  if (globelObj?.ExpoModules?.ExpoRandom) {
> +    const bytes = globelObj.ExpoRandom.getRandomBase64String(BYTE_LENGTH);
>  
>      return bytes;
>    }
>  
> -  if ((global as any).RNPkceChallenge) {
> -    const bytes = (global as any).RNPkceChallenge.randomBytes(BYTE_LENGTH);
> +  if (globelObj?.RNPkceChallenge) {
> +    const bytes = globelObj.RNPkceChallenge.randomBytes(BYTE_LENGTH);
>  
>      return bytes;
>    }
> ```
> 
> _This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296)._

saved my day, this should be implemented!

**Comment by dcangulo** [comment]
This should be fixed in [v5.2.0](https://github.com/dcangulo/react-native-pkce-challenge/releases/tag/v5.2.0).
